### ISSUE TITLE
top-k support for dim 2 sizes {128, 1024, 2048}

### ISSUE THREAD

**Issue by sraizada-tt** [tenstorrent/tt-metal#9006]
unit test: 
```
def test_top_k(device):
    gate_logits_1SBE= ttnn.from_torch(
            torch.zeros(1, 1, 128, 64),
            dtype=ttnn.bfloat16,
            layout=ttnn.TILE_LAYOUT,
            device=device,
        )
    ttl_topk_values, ttl_topk_indices = ttnn.experimental.operations.primary.topk(gate_logits_1SBE, 32)

    print(ttl_topk_values, ttl_topk_indices)
```
